docs/how-to-guides/construct-test-data.md

```diff
@@ -0,0 +1,96 @@
+
+**Limitations**
+
+- While the test data generator works well with standard documents, it may face challenges with API introduction documents or reference documents.
+- Additionally, it may not function effectively for non-Latin characters, such as Chinese. These limitations may be due to the text loader capabilities such as pypdf.
+
+## Quick start
+- Install required packages
+Enter `test_data_gen_local` folder, run below command to install required packages.
+  ```bash
+  pip install -r requirements.txt
+  ```
+
+  For specific document file types, you will need to install extra packages:
+  > !Note: This package requirement may be outdated. In this process, we utilize llama index `SimpleDirectoryReador`. For the most recent information, please check [here](https://docs.llamaindex.ai/en/stable/examples/data_connectors/simple_directory_reader.html).
+  - .docx - `pip install docx2txt`
+  - .pdf - `pip install pypdf`
+  - .ipynb - `pip install nbconvert`
+
+- Run and tune test data construction flow
+  - Navigate to the [construct_test_data_flow folder](../../examples/test_data_gen/construct_test_data_flow/). Open the `flow.dag.yaml` to understand the structure of the data flow. Fill in necessary values like connections and model/deployment name.
+  - Ensure the `flow.tools.json` is generated under `.promptflow` folder within the flow folder.
+  - Check the guide [init-and-test-a-flow](https://microsoft.github.io/promptflow/how-to-guides/init-and-test-a-flow.html) to test the flow.
+  - [*Optional*] Customize your test data generation logic. Refer to [tune-prompts-with-variants](https://microsoft.github.io/promptflow/how-to-guides/tune-prompts-with-variants.html) for more guidance. Once you updated the flow, ensure the flow can complete successfully before preceding to next steps to generate bulk data.
+  
+  **Understand the prompts**
+  
+  The test data construction flow contains five different prompts, classified into two categories based on their roles: generation prompts and validation prompts. Generation prompts are used to create questions, ground truths, etc., while validation prompts are used to verify the validity of the text trunk, generated question or ground truth.
+  - Generation prompts
+    - *generate question prompt*: frame a question based on the given text trunk.
+    - *generate ground truth prompt*: generate ground truth for the question based on the given text trunk.
+  - Validation prompts
+    - *validate text trunk prompt*: validate if the given text trunk is worthy of framing a question.
+    - *validate seed/test question prompt*: validate if the generated question can be clearly understood.
+    - *validate ground truth*: validate if the generated ground truth is clear and certain.
+
+    If the validation fails, the corresponding output would be an empty string so that the invalid data would not be incorporated into the final test data set.
+  
+  **Set the appropriate model and corresponding response format**:
```

Comment:
This should be in the step that you ask user to fill in connection/model/deployment_name